core/src/main/resources/error/error-classes.json

```diff
@@ -346,6 +346,18 @@
       }
     }
   },
+  "INDEX_ALREADY_EXISTS" : {
+    "message" : [
+      "Cannot create the index because it already exists. <message>."
+    ],
+    "sqlState" : "42000"
+  },
+  "INDEX_NOT_FOUND" : {
+    "message" : [
+      "Cannot find the index. <message>."
```

Comment:
What's the `message`? Can't you create sub-classes instead?

Reply:
@MaxGekk I would love to. However this one is a wrapper for JDBC. The message can be anything depending on the data source.
We will need a strategy on how to deal with these cases.

Reply:
We convert `SQLException` in such way as I can see at:
https://github.com/apache/spark/blob/34d5272663ce4852ca5b2daa665983a321b42060/sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala#L223-L232

Cann't we just pass the sqlState and `sqlException.getErrorCode` + vendor name. The info should be enough to identify the error, not?

Reply:
> Can we do this later?

Yep, sure.